http binding - updated documentation (#3369)
dapr/docs: daprdocs/content/en/reference/components-reference/supported-bindings/http.md

- If the pem encoded string is provided, the string is used as is.
When these fields are configured, the Dapr sidecar uses the provided certificate to authenticate itself with the server during the TLS handshake process.

> If remote server is enforcing TLS renegotiation you need to set also metadata field `MTLSRenegotiation`, it accepts one of following options: `RenegotiateNever`,`RenegotiateOnceAsClient` or `RenegotiateFreelyAsClient`. For more details check please: https://pkg.go.dev/crypto/tls#RenegotiationSupport

Reviewer: ```suggestion
If the remote server is enforcing TLS renegotiation, you also need to set the metadata field `MTLSRenegotiation`. This field accepts one of following options: 
- `RenegotiateNever`
- `RenegotiateOnceAsClient` 
- `RenegotiateFreelyAsClient`. 

For more details, see [the Go `RenegotiationSupport` documentation](https://pkg.go.dev/crypto/tls#RenegotiationSupport).
```

Author: @hhunter-ms suggested changes added
